testutils/file_utils.go

```diff
@@ -25,3 +27,37 @@ func BuildTempModule(tb testing.TB, dir string) (string, error) {
 	}
 	return modPath, nil
 }
+
+// MockBuffer is a buffered writer that just appends data to an array to read
+// without needing a real file system for testing.
+type MockBuffer struct {
+	lock   sync.Mutex
+	Writes []*v1.SensorData
+}
+
+// Write adds a collected sensor reading to the array.
+func (m *MockBuffer) Write(item *v1.SensorData) error {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	m.Writes = append(m.Writes, item)
+	return nil
+}
+
+// Flush does nothing in this implementation as all data will be stored in memory.
+func (m *MockBuffer) Flush() error {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	return nil
+}
+
+// Path returns a hardcoded fake path.
+func (m *MockBuffer) Path() string {
+	return "mock dir"
```

Comment:
[supernit] just in case the path ends up being validated later, can make this a valid path, e.g. `/mock/dir`